packages/web-pkg/src/components/CreateLinkModal.vue

```diff
@@ -0,0 +1,427 @@
+    )
+    const onlyInternalLinksAllowed = computed(
+      () => unref(availableRoleOptions).length === 1 && unref(selectedRoleIsInternal)
+    )
+
+    const selectedExpiryDateRelative = computed(() =>
+      formatRelativeDateFromDateTime(
+        DateTime.fromJSDate(unref(selectedExpiry)).endOf('day'),
+        currentLanguage
+      )
+    )
+
+    const selectedExpiryDate = computed(() =>
+      formatRelativeDateFromDateTime(
+        DateTime.fromJSDate(unref(selectedExpiry)).endOf('day'),
+        currentLanguage
+      )
+    )
+
+    const expirationDateTooltip = computed(() => {
+      return $gettext(
+        'Expires %{timeToExpiry} (%{expiryDate})',
+        { timeToExpiry: unref(selectedExpiryDateRelative), expiryDate: unref(selectedExpiryDate) },
+        true
+      )
+    })
+
+    const createLinks = () => {
+      return loadingService.addTask(() =>
+        Promise.allSettled<Share>(
+          props.resources.map((resource) => {
+            const params = {
+              name: $gettext('Link'),
+              expireDate: unref(selectedExpiry),
+              password: unref(password).value,
+              permissions: unref(selectedRole).bitmask(false),
+              spaceRef: resource.fileId,
+              storageId: resource.fileId || resource.id
+            }
+
+            let path = resource.path
+            // sharing a share root from the share jail -> use resource name as path
+            if (resource.isReceivedShare() && path === '/') {
+              path = `/${resource.name}`
+            }
+
+            return store.dispatch('Files/addLink', {
+              path: resource.path,
+              client: clientService.owncloudSdk,
+              params,
+              storageId: resource.fileId || resource.id
+            })
+          })
+        )
+      )
+    }
+
+    const confirm = async () => {
+      if (unref(passwordEnforced) && !unref(password).value) {
+        password.error = $gettext('Password must not be empty')
+        return
+      }
+
+      if (!passwordPolicy.check(unref(password).value)) {
+        return
+      }
+
+      const result = await createLinks()
+
+      const succeeded = result.filter(
+        (val): val is PromiseFulfilledResult<Share> => val.status === 'fulfilled'
+      )
+      if (succeeded.length) {
+        store.dispatch('showMessage', {
+          title:
+            succeeded.length > 1
+              ? $gettext('Links have been created successfully')
+              : $gettext('Link has been created successfully')
+        })
+
+        if (unref(isEmbedEnabled)) {
+          postMessage<string[]>(
+            'owncloud-embed:share',
+            succeeded.map(({ value }) => value.url)
+          )
+        }
+      }
+
+      const failed = result.filter(({ status }) => status === 'rejected')
+      if (failed.length) {
+        failed.forEach((error) => {
+          console.error(error)
+          store.dispatch('showErrorMessage', {
```

Comment:
a bit weird to have a generic, pluralized message for each and every error individually. 😅